src/dmclock_server.h

```diff
@@ -783,15 +787,34 @@ namespace crimson {
       {
 	assert(_erase_age >= _idle_age);
 	assert(_check_time < _idle_age);
-	cleaning_job =
+	idle_erase_job =
 	  std::unique_ptr<RunEvery>(
 	    new RunEvery(check_time,
-			 std::bind(&PriorityQueueBase::do_clean, this)));
+			 std::bind(&PriorityQueueBase::do_idle_erase, this)));
       }
 
 
       ~PriorityQueueBase() {
 	finishing = true;
+
+	DataGuard g(data_mtx);
+
+	ready_heap.clear();
+	limit_heap.clear();
+#if USE_PROP_HEAP
+	prop_heap.clear();
+#endif
+	resv_heap.clear();
+
+	for (auto c = client_map.begin(); client_map.end() != c; /* empty */) {
+	  auto current = c++;
+	  for (auto l : idle_erase_listeners) {
+	    (*l)(current->second->client);
```

Comment:
what's the motivation for notifying these listeners on destruction?

Reply:
As I think about this further, not allowing the listener to be dynamic constrains the owner but likely makes more sense overall.